solidity_contracts/GetCallerAddress.sol

```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract GetCallerAddress {
+    // This function will return the address of the caller
+    function getCallerAddress() public view returns (address) {
+        return msg.sender;
```

Comment:
this should return 0 or any caller address, right? we're just checking that it doesnt error out

> It appears you're running a call, which runs quickly and does not alter the state of the blockchain. msg.sender is set in both a transaction and a call. In a transaction, it cannot be faked: you must have the private key associated with the given account. But in a call, you are free to set the sender to any value you like.

from:
https://stackoverflow.com/questions/51847788/msg-sender-does-not-work-inside-a-view-function-why-is-there-a-workaround